### PR TITLE
My code review suggestions

### PR DESCRIPTION
**Pull request by bhagwat-rahul**
Did the app compile: yes
Description: This is an activity tracker that tracks activities by name, saves and displays them as a pie chart.
What I learned: 
Best practices for kotlin programming
Naming conventions
File and package hierarchy
Writing clean easy to understand code

(P.S:- Out of all the projects I have seen, in my opinion this one was the best in terms of code readability and ease of use)

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -9,9 +9,9 @@
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
         android:theme="@style/AppTheme">
-        <activity android:name=".ViewActivityHistory"></activity>
+        <activity android:name=".ui.ViewActivityHistory"></activity>
         <activity
-            android:name=".MainActivity"
+            android:name=".ui.MainActivity"
```
Comment (bhagwat-rahul) [inline app/src/main/AndroidManifest.xml:14]: created a new package for mainactivity to provide more structure and hierarchy to the directory

File: app/src/main/java/com/example/timemanagementapp/ui/MainActivity.kt [hunk 2]
```diff
@@ -20,7 +21,7 @@ import kotlin.collections.ArrayList
 
 class MainActivity : AppCompatActivity() {
 
-    val DEFAULTACTIVITYNAME = "UC2020_DefaultActivity"
+    val DEFAULT_ACTIVITY_NAME = "UC2020_DefaultActivity"
```
Comment (bhagwat-rahul) [inline app/src/main/java/com/example/timemanagementapp/ui/MainActivity.kt:25]: immutable val's should have capitalized names with underscores as per kotlin best practices.

https://kotlinlang.org/docs/reference/coding-conventions.html

File: app/src/main/java/com/example/timemanagementapp/ui/MainActivity.kt [hunk 3]
```diff
@@ -69,9 +70,11 @@ class MainActivity : AppCompatActivity() {
...
 
     override fun onOptionsItemSelected(item: MenuItem): Boolean {
-        // Handle action bar item clicks here. The action bar will
-        // automatically handle clicks on the Home/Up button, so long
-        // as you specify a parent activity in AndroidManifest.xml.
+        /**
+         *  Handle action bar item clicks here. The action bar will
+         *  automatically handle clicks on the Home/Up button, so long
+         *  as you specify a parent activity in AndroidManifest.xml.
+         */
```
Comment (bhagwat-rahul) [inline app/src/main/java/com/example/timemanagementapp/ui/MainActivity.kt:77]: The structure of multiline comments should be such as per convention

https://kotlinlang.org/docs/reference/coding-conventions.html

File: app/src/main/java/com/example/timemanagementapp/ui/MainActivity.kt [hunk 4]
```diff
@@ -88,7 +91,8 @@ class MainActivity : AppCompatActivity() {
             val currentDateFile = File(currentDateDirectoryName)
             val gson = Gson()
 
-            val activityLogFilePath = File("$currentDateFile/$currentDate.json")
+            val activityLogFilePath =
+                File("$currentDateFile/$currentDate.json")
```
Comment (bhagwat-rahul) [inline app/src/main/java/com/example/timemanagementapp/ui/MainActivity.kt:95]: for properties with long initializers add a line break after '=' sign for better readability

https://kotlinlang.org/docs/reference/coding-conventions.html

File: app/src/main/java/com/example/timemanagementapp/ui/MainActivity.kt [hunk 5]
```diff
@@ -109,8 +113,9 @@ class MainActivity : AppCompatActivity() {
              var dailyActivityLog: DailyActivityLogDTO = gson.fromJson(jsonString, DailyActivityLogDTO::class.java)
 
              // using a CopyOnWriteArrayList prevents an error where more than one thread is operating on the list, leading to an error
-             val syncedActivityLog =  CopyOnWriteArrayList(dailyActivityLog.activityLog)
-             val syncedActivityNames = CopyOnWriteArrayList<String>();
+             val syncedActivityLog =
+                 CopyOnWriteArrayList(dailyActivityLog.activityLog)
```
Comment (bhagwat-rahul) [inline app/src/main/java/com/example/timemanagementapp/ui/MainActivity.kt:117]: for properties with long initializers add a line break after '=' sign for better readability

https://kotlinlang.org/docs/reference/coding-conventions.html

File: app/src/main/java/com/example/timemanagementapp/ui/MainActivity.kt [hunk 6]
```diff
@@ -143,7 +148,8 @@ class MainActivity : AppCompatActivity() {
         }
 
         fun viewActivityHistory(view: View){
-            val intent = Intent(this, ViewActivityHistory::class.java)
+            val intent =
+                Intent(this, ViewActivityHistory::class.java)
```
Comment (bhagwat-rahul) [inline app/src/main/java/com/example/timemanagementapp/ui/MainActivity.kt:152]: for properties with long initializers add a line break after '=' sign for better readability

https://kotlinlang.org/docs/reference/coding-conventions.html

File: app/src/main/java/com/example/timemanagementapp/ui/MainActivity.kt [hunk 7]
```diff
@@ -127,7 +132,7 @@ class MainActivity : AppCompatActivity() {
              if(isNameDuplicated){
                  var indexOfDuplicatedName = syncedActivityNames.indexOf(newActivityName.toString())
                  var duplicateActivity = dailyActivityLog.activityLog.get(indexOfDuplicatedName)
-                 dailyActivityLog.activityLog.removeAt(indexOfDuplicatedName);
+                 dailyActivityLog.activityLog.removeAt(indexOfDuplicatedName)
```
Comment (bhagwat-rahul) [inline app/src/main/java/com/example/timemanagementapp/ui/MainActivity.kt:135]: unnecessary

**Review by bhagwat-rahul — COMMENTED** [review]
my revieiw summary:
comment structure
naming 'vals'
creating appropriate packages for classes
long initializers on next line

File: app/src/main/java/com/example/timemanagementapp/ui/MainActivity.kt (hunk 2 above)
Comment (discospiff) [reply app/src/main/java/com/example/timemanagementapp/ui/MainActivity.kt:25]: I agree... and good citation on naming conventions!

File: app/src/main/java/com/example/timemanagementapp/ui/MainActivity.kt (hunk 3 above)
Comment (discospiff) [reply app/src/main/java/com/example/timemanagementapp/ui/MainActivity.kt:77]: I recommend one more change... JavaDoc comments belong above a function, so move the /** */ comment above the function signature line, override fun onOptionsItemSelected.

File: app/src/main/java/com/example/timemanagementapp/ui/MainActivity.kt (hunk 7 above)
Comment (discospiff) [reply app/src/main/java/com/example/timemanagementapp/ui/MainActivity.kt:135]: Old habits can be hard to break... I still catch myself ending lines with ; periodically.  :)

File: app/src/main/java/com/example/timemanagementapp/ui/MainActivity.kt (hunk 6 above)
Comment (discospiff) [reply app/src/main/java/com/example/timemanagementapp/ui/MainActivity.kt:152]: This one is not too long, so safe to keep on one line.  I find these breaks to be more important:
1) When you have a line that is longer than 80 characters, or
2) When you are chaining multiple method/function calls together.

**Comment by discospiff** [comment]
Several good suggestions here.  See inline comments for more details.  I recommend that the group review each change, and decide whether to integrate certain changes manually into their project, or merge the entire branch.

**Pull request closed without merging**